tests/test-callbacks.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2017, Intel Corporation.
+
+var assert = require("Assert.js");
+var cb = require('test_callbacks');
+
+var times = 0;
+function callback(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "a prop"),
+            "args matched (" + times + ")");
+    times++;
+}
+
+var list_count = 0;
+function callback_l1(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "list"),
+            "list args matched (" + list_count + ")");
+    list_count++;
+}
+
+function callback_l2(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "list"),
+            "list args matched (" + list_count + ")");
+    list_count++;
+}
+
+function callback_l3(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "list"),
+            "list args matched (" + list_count + ")");
+    list_count++;
+}
+
+var id = cb.addCallback(callback, {prop: "a prop"});
+cb.signalCallback(id, "one", "two", "three");
+cb.signalCallback(id, "one", "two", "three");
+cb.signalCallback(id, "one", "two", "three");
+
+setTimeout(function() {
+    assert(times === 3, "signaled 3 times");
+    cb.removeCallback(id);
+    cb.signalCallback(id, "one", "two", "three");
+    setTimeout(function() {
+        assert(times === 3, "removed callback not called");
+        cb.signalCallback(-1);
+        cb.signalCallback(-10);
+        cb.signalCallback(12345);
+        setTimeout(function() {
+            assert(true, "with bad ID's didn't crash");
```

Comment:
Since its the main loop (service_callbacks()) which actually calls the callback that may throw an error there is no way we can catch it unfortunately. All you get is a print that an error happened. This is a limitation of JerryScript which would be nice to get figured out but for now its not possible.

And no you cant add a callback list to another list.